app/models/custom_crawler.rb

```diff
@@ -0,0 +1,25 @@
+class CustomCrawler
+  require 'digest/md5'
+
+  def initialize
+    option_hash = { crawl_limit_by_page: true }
+    @cw = Cobweb.new(option_hash)
+  end
+
+  def recursive_get(base_url, depth = 2)
+    unless depth < 0
+      begin
+        f = @cw.get(base_url)
+      rescue
+        puts "a connection was refused, move on"
```

Comment:
Instead of using `puts` here, you should use the [Rails logger] with the appropriate log level.  For example, `Rails.logger.info "Connection refused, moving on"`.  Also, generally errors are written in present tense. :memo: